mits/mits_suite_test.go

```diff
@@ -17,45 +17,83 @@
 package mits_test
 
 import (
-	"fmt"
-	"log"
 	"os"
 	"testing"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
 
-	"gopkg.in/yaml.v2"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/cf"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/config"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/generator"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/workflowhelpers"
 
 	"github.com/SUSE/minibroker-integration-tests/mits"
 )
 
-var testsConfig mits.TestsConfig
+var (
+	tests    mits.TestsConfig
+	timeouts mits.TimeoutsConfig
 
-func init() {
-	if err := loadConfig("CONFIG_TESTS", &testsConfig); err != nil {
-		log.Fatal(err)
-	}
-}
-
-func loadConfig(envConfig string, config interface{}) error {
-	configPath, ok := os.LookupEnv(envConfig)
-	if !ok {
-		return fmt.Errorf("failed to load config: %q environment variable is not set", envConfig)
-	}
-	configFile, err := os.Open(configPath)
-	if err != nil {
-		return fmt.Errorf("failed to load config: %w", err)
-	}
-	defer configFile.Close()
-	configDecoder := yaml.NewDecoder(configFile)
-	if err := configDecoder.Decode(config); err != nil {
-		return fmt.Errorf("failed to load config: %w", err)
-	}
-	return nil
-}
+	testSetup         *workflowhelpers.ReproducibleTestSuiteSetup
+	serviceBrokerName string
+	serviceBrokerURL  string
+)
 
 func TestMits(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "Mits Suite")
 }
+
+var _ = BeforeSuite(func() {
+	err := mits.LoadConfig("CONFIG_TESTS", &tests)
+	Expect(err).NotTo(HaveOccurred())
+	err = mits.LoadConfig("CONFIG_TIMEOUTS", &timeouts)
+	Expect(err).NotTo(HaveOccurred())
+
+	apiEndpoint := os.Getenv("API_ENDPOINT")
+	adminUser := os.Getenv("CF_ADMIN_USERNAME")
+	adminPassword := os.Getenv("CF_ADMIN_PASSWORD")
```

Comment:
Would it make sense to put _everything_ in a YAML file, rather than 2 files + 4 vars?

Reply:
Happily done with 243e8ad.